core/src/test/java/com/google/errorprone/ErrorProneJavaCompilerTest.java

```diff
@@ -486,6 +486,32 @@ class StringConstantWrapper {
             """);
   }
 
+  @Test
+  public void patchAllWithDisableAllChecks() throws IOException {
+    JavaFileObject fileObject =
+        createOnDiskFileObject(
+            "StringConstantWrapper.java",
+            """
+            class StringConstantWrapper {
+              String s = "old-value";
+            }
+            """);
+
+    CompilationResult result =
+        doCompile(
+            Collections.singleton(fileObject),
+            Arrays.asList("-XepPatchChecks:", "-XepPatchLocation:IN_PLACE", "-XepDisableAllChecks"),
+            ImmutableList.of(AssignmentUpdater.class));
+    assertThat(result.succeeded).isTrue();
+    assertThat(Files.readString(Path.of(fileObject.toUri())))
+        .isEqualTo(
+            """
+            class StringConstantWrapper {
+              String s = "old-value";
+            }
+            """);
+  }
```

Comment:
This new test fails prior to the proposed changes. Observe that the only difference with the preceding `patchAllWithCheckDisabled` test is the use of `-XepDisableAllChecks` instead of `-Xep:AssignmentUpdater:OFF`.